### PR TITLE
[wgsl] Add validation test - v-0032: constraints on module variables with an initializers

### PR DESCRIPTION
**Pull request by sarahM0**
If a variable has an initializer, then its storage class must be one of private, function, or out.
[Link](https://gpuweb.github.io/gpuweb/wgsl.html#variables) to WGSL spec section citing this rule.

-----

**[Reviews requirements](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) are satisfied for:**

- [x] WebGPU readability
- [ ] TypeScript readability

### CODE REVIEW

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-module-workgroup.fail.wgsl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# v-0032: variable 'w' has an initializer, however its storage class is 'workgroup'.
+
+var<workgroup> w : vec3<i32> = vec3<i32>(0, 1, 0);
```
Comment (sarahM0) [inline src/webgpu/shader/validation/wgsl/variable-with-initilizer-module-workgroup.fail.wgsl:3]: Is this a correct use of <wrokgoup> storage class or should the store type be 'array'?
Is there a way to initialize arrays in wgsl/tint?

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-module-uniform.fail.wgsl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# v-0032: variable 'u' has an initializer, however its storage class is 'uniform'.
+
+[[block]] struct Params {
+  [[offset(0)]] count: i32;
+};
+var<uniform> u : Params = 1;
```
Comment (sarahM0) [inline src/webgpu/shader/validation/wgsl/variable-with-initilizer-module-uniform.fail.wgsl:6]: A variable in 'uniform' storage class has to be a struct. Is there a way to init a struct?

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# v-0032: variable 'u' has an initializer, however its storage class is 'storage'.
+
+[[block]]
+struct PositionBuffer {
+  [[offset(0)]] pos: vec2<f32>; 
+};
+
+var<storage_buffer> s : PositionBuffer = vec2<f32>(0.0, 0.0);
```
Comment (sarahM0) [inline src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl:8]: possible spec issue: I cannot find one use of "storage_buffer" in spec. Should it be "var<storage> s" instead.

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# v-0032: variable 'u' has an initializer, however its storage class is 'storage'.
+
+[[block]]
+struct PositionBuffer {
+  [[offset(0)]] pos: vec2<f32>; 
+};
+
+[[group(0), binding(0)]]
+var<storage_buffer> s : PositionBuffer = {vec2<f32>(0.0, 0.0)}; 
+
```
Comment (sarahM0) [inline src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl:10]: tint ouput:

variable-with-initilizer-storage.fail.wgsl:8:3 error: expected decoration
[[group(0), binding(0)]]
  ^^^^^

variable-with-initilizer-storage.fail.wgsl:9:42 error: unable to parse const literal
var<storage_buffer> s : PositionBuffer = {vec2<f32>(0.0, 0.0)};

Comment (dj2) [reply src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl:10]: This is a Tint bug, we're still parsing `set` and not `group` yet.

**Review by dj2 — APPROVED** [review]

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# v-0032: variable 'u' has an initializer, however its storage class is 'storage'.
+
+[[block]]
+struct PositionBuffer {
+  [[offset(0)]] pos: vec2<f32>; 
+};
+
+[[group(0), binding(0)]]
```
Comment (sarahM0) [inline src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl:8]: I'm probably missing something, tint ouput:
variable-with-initilizer-storage.fail.wgsl:8:3 error: expected decoration
[[group(0), binding(0)]]
  ^^^^^

Comment (dj2) [reply src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl:8]: This is tint falling behind the spec, tint is still looking for `set` not `group`. So, this is correct and Tint needs to be updated.

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl (hunk 3 above)
Comment (dj2) [reply src/webgpu/shader/validation/wgsl/variable-with-initilizer-storage.fail.wgsl]: Correct, this was renamed to `storage` and tint hasn't caught up yet. (See https://github.com/gpuweb/gpuweb/pull/1093/files for the name update)

File: src/webgpu/shader/validation/wgsl/variable-with-initilizer-module-workgroup.fail.wgsl (hunk 1 above)
Comment (dj2) [reply src/webgpu/shader/validation/wgsl/variable-with-initilizer-module-workgroup.fail.wgsl]: I believe this is correct.

**Review by dj2 — APPROVED** [review]

**Pull request merged**